api/numaresourcesoperator/v1/helper/nodegroup/nodegroup.go

```diff
@@ -106,9 +106,23 @@ func FindTrees(mcps *mcov1.MachineConfigPoolList, nodeGroups []nropv1.NodeGroup)
 	return result, nil
 }
 
+func FindTreesHypershift(nodeGroups []nropv1.NodeGroup) []Tree {
```

Comment:
since the slice size of tree and nodegroup  are equal we can optimize a little bit:
```
func FindTreesHypershift(nodeGroups []nropv1.NodeGroup) []Tree {
	result := make([]Tree, len(nodeGroups))
	for i := range nodeGroups {
		result[i] = Tree{NodeGroup: &nodeGroups[i]}
	}
	return result
}
```

Reply:
update, thanks